java-connectors/kafka-connect-gcp-common/src/test/java/io/lenses/streamreactor/connect/gcp/common/config/GCPSettingsTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.gcp.common.config;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.lenses.streamreactor.common.config.base.RetryConfig;
+import io.lenses.streamreactor.common.config.base.model.ConnectorPrefix;
+import io.lenses.streamreactor.common.config.source.MapConfigSource;
+import java.util.Map;
+import java.util.stream.Stream;
+import lombok.val;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class GCPSettingsTest {
+
+  private static final ConnectorPrefix connectorPrefix = new ConnectorPrefix("connect.gcpstorage");
+  private static final GCPSettings gcpSettings = new GCPSettings(connectorPrefix);
+
+  private static Stream<Arguments> provideRetryConfigData() {
+    return Stream.of(
+        Arguments.of("no values", 0, 0L, new RetryConfig(0, 0L)),
+        Arguments.of("retry limit only", 10, 0L, new RetryConfig(10, 0L)),
+        Arguments.of("interval only", 0, 20L, new RetryConfig(0, 20L)),
+        Arguments.of("retry limit and interval", 30, 40L, new RetryConfig(30, 40L)));
+  }
+
+  @ParameterizedTest(name = "{0}")
```

Comment:
MINOR: It's valid to do it via method, but here we could maybe just have a static Stream of arguments like [here](https://www.baeldung.com/parameterized-tests-junit-5#8-custom-annotation)? It's save us from `{0}` ;)

Reply:
Apologies, I've looked everywhere, even under the kitchen sink, but I cannot find a `@VariableSource`.

Reply:
The article you linked suggests creating a custom annotation to support the `VariableSource`.

To quote the article:
> Suppose we want to load the test arguments from a static variable:
> Actually, JUnit 5 does not provide this. However, we can roll our own solution.
> 
> First, we can create an annotation:
> ```
> @Documented
> @Target(ElementType.METHOD)
> @Retention(RetentionPolicy.RUNTIME)
> @ArgumentsSource(VariableArgumentsProvider.class)
> public @interface VariableSource {
> ```


So I will leave it as it is, as the effort is too great for minimal benefit.

Reply:
Got it, I think I made it a step too far then, sorry!